roboteam_ai/src/skills/GoToPos.cpp

```diff
@@ -30,19 +27,17 @@ void GoToPos::onInitialize() {
 
 /// Get an update on the skill
 bt::Node::Status GoToPos::onUpdate() {
-    updateRobot();
     if (! robot) return Status::Running;
+    if (!ball) return Status::Running;
```

Comment:
We do not want to check for the ball if the GoToPos does not need the ball to compute its path. Surround this with an if (goToBall or goBehindBall) statement.